src/go/configgenerator/route_generator.go

```diff
@@ -255,100 +255,116 @@ func makeRouteTable(serviceInfo *configinfo.ServiceInfo) ([]*routepb.Route, erro
+							NumRetries: &wrapperspb.UInt32Value{
+								Value: uint32(method.BackendInfo.RetryNum),
+							},
 						},
 					},
 				},
-			},
-			Decorator: &routepb.Decorator{
-				// Note we don't add ApiName to reduce the length of the span name.
-				Operation: fmt.Sprintf("%s %s", util.SpanNamePrefix, method.ShortName),
-			},
-		}
+				Decorator: &routepb.Decorator{
+					// Note we don't add ApiName to reduce the length of the span name.
+					Operation: fmt.Sprintf("%s %s", util.SpanNamePrefix, method.ShortName),
+				},
+			}
 
-		r.TypedPerFilterConfig, err = makePerRouteFilterConfig(operation, method, httpRule)
-		if err != nil {
-			return nil, fmt.Errorf("fail to make per-route filter config, %v", err)
-		}
+			r.TypedPerFilterConfig, err = makePerRouteFilterConfig(operation, method, httpRule)
+			if err != nil {
+				return nil, fmt.Errorf("fail to make per-route filter config, %v", err)
+			}
 
-		if method.BackendInfo.Hostname != "" {
-			// For routing to remote backends.
-			r.GetRoute().HostRewriteSpecifier = &routepb.RouteAction_HostRewriteLiteral{
-				HostRewriteLiteral: method.BackendInfo.Hostname,
+			if method.BackendInfo.Hostname != "" {
+				// For routing to remote backends.
+				r.GetRoute().HostRewriteSpecifier = &routepb.RouteAction_HostRewriteLiteral{
+					HostRewriteLiteral: method.BackendInfo.Hostname,
+				}
 			}
-		}
 
-		if serviceInfo.Options.EnableHSTS {
-			r.ResponseHeadersToAdd = []*corepb.HeaderValueOption{
-				{
-					Header: &corepb.HeaderValue{
-						Key:   util.HSTSHeaderKey,
-						Value: util.HSTSHeaderValue,
+			if serviceInfo.Options.EnableHSTS {
+				r.ResponseHeadersToAdd = []*corepb.HeaderValueOption{
+					{
+						Header: &corepb.HeaderValue{
+							Key:   util.HSTSHeaderKey,
+							Value: util.HSTSHeaderValue,
+						},
 					},
-				},
+				}
 			}
-		}
-		backendRoutes = append(backendRoutes, &r)
+			backendRoutes = append(backendRoutes, &r)
 
-		jsonStr, _ := util.ProtoToJson(&r)
-		glog.Infof("adding route: %v", jsonStr)
+			jsonStr, _ := util.ProtoToJson(&r)
+			glog.Infof("adding route: %v", jsonStr)
+		}
 	}
 	return backendRoutes, nil
 }
 
-func makeHttpRouteMatcher(httpRule *httppattern.Pattern) (*routepb.RouteMatch, error) {
+func makeHttpExactPathRouteMatcher(path string) *routepb.RouteMatch {
+	return &routepb.RouteMatch{
+		PathSpecifier: &routepb.RouteMatch_Path{
+			Path: path,
+		},
+	}
+}
+
+func makeHttpRouteMatchers(httpRule *httppattern.Pattern) ([]*routepb.RouteMatch, error) {
 	if httpRule == nil {
 		return nil, fmt.Errorf("httpRule is nil")
 	}
-	var routeMatcher routepb.RouteMatch
+	var routeMatchers []*routepb.RouteMatch
 
 	if httpRule.UriTemplate.IsExactMatch() {
-		routeMatcher = routepb.RouteMatch{
-			PathSpecifier: &routepb.RouteMatch_Path{
-				Path: httpRule.UriTemplate.String(),
-			},
+		pathNoTrailingSlash := httpRule.UriTemplate.ExactMatchString(false)
+		pathWithTrailingSlash := httpRule.UriTemplate.ExactMatchString(true)
```

Comment:
DITTO

Reply:
See above. Let's keep it as a boolean for now, if we need to support more args, we can move into a struct for readability.